Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -190,8 +202,27 @@ void TRDGlobalTracking::run(ProcessingContext& pc)
   mChainTracking->ClearIOPointers();
   o2::globaltracking::RecoContainer inputTracks;
   inputTracks.collectData(pc, *mDataRequest);
+  mTPCClusterIdxStruct = &inputTracks.inputsTPCclusters->clusterIndex;
+  mTPCRefitter = std::make_unique<o2::gpu::GPUO2InterfaceRefit>(mTPCClusterIdxStruct, mTPCTransform.get(), o2::base::Propagator::Instance()->getNominalBz(), inputTracks.getTPCTracksClusterRefs().data(), inputTracks.clusterShMapTPC.data(), nullptr, o2::base::Propagator::Instance());
   auto tmpInputContainer = getRecoInputContainer(pc, &mChainTracking->mIOPtrs, &inputTracks, mUseMC);
   auto tmpContainer = GPUWorkflowHelper::fillIOPtr(mChainTracking->mIOPtrs, inputTracks, mUseMC, nullptr, GTrackID::getSourcesMask("TRD"), mTrkMask, GTrackID::mask_t{GTrackID::MASK_NONE});
+  mTrackletsRaw = inputTracks.getTRDTracklets();
+  mTrackletsCalib = inputTracks.getTRDCalibratedTracklets();
+  mTPCTracksArray = inputTracks.getTPCTracks();
+  if (GTrackID::includesDet(GTrackID::DetID::ITS, mTrkMask)) {
+    mITSTracksArray = inputTracks.getITSTracks();
+    auto mITSClusterROFRec = inputTracks.getITSClustersROFRecords();
+    const auto clusITS = inputTracks.getITSClusters();
+    if (mITSClusterROFRec.empty() || clusITS.empty()) {
+      LOG(FATAL) << "No ITS clusters";
+      return;
+    }
+    const auto patterns = inputTracks.getITSClustersPatterns();
+    auto pattIt = patterns.begin();
+    mITSClustersArray.reserve(clusITS.size());
+    //o2::its::ioutils::convertCompactClusters(clusITS, pattIt, mITSClustersArray, *mITSDict); // segfaults.. mITSDict is not set, is this the issue?
```

Comment:
Hi @shahor02 do know what I am doing wrong here trying to load the ITS clusters?

Reply:
ITSDict must be a valid object, even if empty (i.e. created by default constructor). See how it is done here: https://github.com/AliceO2Group/AliceO2/blob/5897718f3ba61f306520709be3815d9e59f94080/Detectors/GlobalTrackingWorkflow/src/TPCITSMatchingSpec.cxx#L101-L107
(eventually it should come from the CCDB, the TPCITSMatchingSpec will update it in the `updateTimeDependentParams` so the MatchTPCITS::mITSDict pointer to TPCITSMatchingSpec::mITSDict will automatically provide the latest dict. In your case, since you keep the fitter in the spec, you can simply convert mITSDict from pointer to object, and read it like in the TPCITSMatchingSpec